lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,492 @@
+"""Initialising Indexing class"""
+
+import os
+import glob
+import signal
+import tarfile
+import tempfile
+from pathlib import Path
+
+from pbench.common.exceptions import (
+    BadDate,
+    UnsupportedTarballFormat,
+    SosreportHostname,
+    BadMDLogFormat,
+    TemplateError,
+)
+from pbench.server import tstos
+from pbench.server.indexer import (
+    PbenchTarBall,
+    es_index,
+    VERSION,
+)
+from pbench.server.report import Report
+from pbench.server.utils import rename_tb_link, quarantine
+
+
+class SigIntException(Exception):
+    pass
+
+
+def sigint_handler(*args):
+    raise SigIntException()
+
+
+class SigTermException(Exception):
```

Comment:
Do we need this here? There is a SigTermException class defined in `pbench-index.py`, together with a signal handler that raises it. So I'm not sure that we need to handle the SigTermException in this module at all, in which case we do not need to define the class. Is this correct?

Reply:
yes. As the exceptions are being raised in this file it is expected by them.

Reply:
Good catch, Nick, I totally missed this subtly.

We should probably keep this exception definition here, and `import` it into the `pbench-index.py` module instead.

Reply:
Oh that, did not cross my mind. yeah we can do that